lightkurve/search.py

```diff
@@ -681,23 +677,13 @@ def open(path_or_url):
 
         >>> tpf = open("mytpf.fits")  # doctest: +SKIP
     """
-    hdulist = fits.open(path_or_url)
-    try:
-        # use `telescop` keyword to determine mission
-        # and `creator` to determine tpf or lc
-        telescop = hdulist[0].header['telescop']
-        creator = hdulist[0].header['creator']
-        if telescop == 'Kepler':
-            if 'TargetPixel' in creator:
-                return KeplerTargetPixelFile(path_or_url)
-            elif 'Flux' in creator:
-                return KeplerLightCurveFile(path_or_url)
-        elif telescop == 'TESS':
-            if 'TargetPixel' in creator:
-                return TessTargetPixelFile(path_or_url)
-            elif 'Flux' in creator:
-                return TessLightCurveFile(path_or_url)
-    # if these keywords don't exist, raise `ValueError`
-    except KeyError:
-        pass
-    raise ValueError('Given fits file not recognized as Kepler or TESS observation.')
+    # pass header into `detect_filetype()`
+    filetype = detect_filetype(fits.open(path_or_url)[0].header)
+
+    # if the filetype is recognized, instantiate a class of that name
+    if filetype is not None:
+        return getattr(__import__('lightkurve'), filetype)(path_or_url)
+    else:
+        # if these keywords don't exist, raise `ValueError`
+        raise ValueError("Given fits file not recognized as Kepler or TESS "
+                         "observation.")
```

Comment:
At this point in the code we're not even sure that `path_or_url` pointed to a fits file, so I'd have this error message say something like `"not recognized as a Kepler or TESS data product: {}".format(path_or_url)`